File: 이것이 취업을 위한 코딩 테스트다/그리디/숫자_카드_게임/숫자_카드_게임.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
  int n, m;
  cin >> n >> m;
  vector<vector<int>>cards;
  int answer = 1;

  for (int i = 0; i < n; i++) {
    int num, min = 10000;
    vector<int> temp;

    for (int j = 0 ; j < m; j++) {
      cin >> num;
      min = num < min ? num : min;
      temp.push_back(num);
    }

    answer = min > answer ? min : answer;

    cards.push_back(temp);
  }

  cout << answer;
}

File: 이것이 취업을 위한 코딩 테스트다/그리디/숫자_카드_게임/숫자_카드_게임.java
package 그리디.숫자_카드_게임;

import java.util.Scanner;

public class 숫자_카드_게임 {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int m = sc.nextInt();

    // int cards[][] = new int[n][m];
    int answer = 0;

    for(int i = 0; i < n; i++) {
      int min = 10000;
      for (int j = 0; j < m; j++) {
        int temp = sc.nextInt();

        min = temp < min ? temp : min;
      }

      answer = answer < min ? min : answer;
    }

    System.out.println(answer);
  }
}
